src/main/java/org/marketdesignresearch/mechlib/core/bid/bundle/BundleValueBid.java

```diff
@@ -0,0 +1,66 @@
+package org.marketdesignresearch.mechlib.core.bid.bundle;
+
+import com.google.common.collect.ImmutableSet;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.ToString;
+
+import org.marketdesignresearch.mechlib.core.BundleEntry;
+import org.marketdesignresearch.mechlib.core.Good;
+import org.marketdesignresearch.mechlib.core.bid.Bid;
+import org.springframework.data.annotation.PersistenceConstructor;
+
+import java.math.BigDecimal;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@RequiredArgsConstructor(onConstructor = @__({@PersistenceConstructor}))
+@ToString @EqualsAndHashCode
+public class BundleValueBid<T extends BundleValuePair> implements Bid{
```

Comment:
Note I changed the nameing here. This is the old Bid. And the BundleBid changed to BundleValuePair. Making this generic enables introducing BundleValueBoundPair easily. However, the handling of the BundleValueBid gets a bit more complicated.